src/agent/block_store_services/block_store_fs.js

```diff
@@ -218,7 +242,7 @@ class BlockStoreFs extends BlockStoreBase {
         if (!is_test_block) {
 
             // set xattr to trigger migration of file to underlying tier
-            if (config.BLOCK_STORE_FS_TMFS_ENABLED) {
+            if (!this.xattr_fallback && config.BLOCK_STORE_FS_TMFS_ENABLED) {
```

Comment:
again, I would leave this out - we don't plan to support it without xattr at the moment, so it would be best to fail if configured with tmfs and xattr is not supported rather than just ignoring the config.

```suggestion
            if (config.BLOCK_STORE_FS_TMFS_ENABLED) {
```